### PR TITLE
Add exceptions for breakage introduced in RD barriers

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This PR makes a few minor changes to https://github.com/godotengine/godot/pull/77420 to:
- make sure the existing enums `BARRIER_MASK_COMPUTE` and `BARRIER_MASK_TRANSFER` keep their original values
- make sure that enums `BARRIER_MASK_ALL_BARRIERS` and `BARRIER_MASK_NO_BARRIER` are more future proof in case we add additional flags in the future.

Added exceptions to 4.0-stable-expected where applicable.

### CODE REVIEW

**Comment by BastiaanOlij** [comment]
Just waiting on CI to finish so we know the breakage exceptions I need to add...

**Review by clayjohn — APPROVED** [review]
Looks good to me. This should minimize compat breakage if anyone is relying on these barriers and it gets rid of our annoying errors

**Review by dsnopek — APPROVED** [review]
This looks good to me as well! Thanks!

File: doc/classes/RenderingDevice.xml
Comment (raulsntos) [inline doc/classes/RenderingDevice.xml:1595]: Old code that used the `BARRIER_MASK_RASTER` flag will be equivalent to using only `BARRIER_MASK_VERTEX` now. Is that OK? Or is the `BARRIER_MASK_FRAGMENT` also important?

**Review by raulsntos — COMMENTED** [review]
I don't really know what these barrier flags are, so I added some comments only regarding compatibility.

It does look like it will break compat less and that looks like a good thing.

File: doc/classes/RenderingDevice.xml
Comment (BastiaanOlij) [reply doc/classes/RenderingDevice.xml:1598]: Yes, all barriers means all barriers, so I think we need to update the documentation to reflect this.

That said, seeing future enhancements will likely purely mean that raster will be broken up further, or maybe that there will be some future reason that compute needs to be divided up it will probably still cover the description.

All in all, the likelihood of further enums being introduced is very tiny, this is purely a matter of "if the value changes anyway, might as well change it to something so we don't get into this situation again, even if that will never happen".

File: doc/classes/RenderingDevice.xml
Comment (BastiaanOlij) [reply doc/classes/RenderingDevice.xml:1595]: Remember we're comparing to what is currently released. So this is a change from "0" to "9", but yes the conclusion is the same. If you used `BARRIER_MASK_RASTER` in an extension in 4.1 and you now run this same extension in 4.2 without recompiling, you will only get a barrier with the vertex shader.

Seeing you can't really run a fragment shader without first running the vertex shader, this is likely safe. 

The original enhancement we're after here is when the vertex shader should not wait for execution before a previous process is finished which was not possible in the past.
As on mobile the vertex shader for all geometry in a pass is run fully before before the fragment shader is applied, this will allow you to run data transfers (populating UBOs) and compute shaders in parallel with the vertex shader. 
On desktop this is a non-issue as vertex and fragment shaders run in parallel.

File: doc/classes/RenderingDevice.xml
Comment (BastiaanOlij) [reply doc/classes/RenderingDevice.xml:19]: Again, remember we're going from the situation in 4.1 to this, the previous PR was only ever part of dev builds.

So "7" was previous raster + compute + transfer, it is now vertex + compute + transfer missing out on the fragment barrier. As my explanation on the other item, this is an acceptable compromise.

Comment (clayjohn) [reply doc/classes/RenderingDevice.xml:19]: I agree, a default of 7 is fine with this PR

File: doc/classes/RenderingDevice.xml
Comment (clayjohn) [reply doc/classes/RenderingDevice.xml:1595]: Just barrier mask vertex is fine!

File: doc/classes/RenderingDevice.xml
Comment (clayjohn) [reply doc/classes/RenderingDevice.xml:1598]: Let's update the docs in this PR before merging

Comment (BastiaanOlij) [reply doc/classes/RenderingDevice.xml:1598]: Updated!

**Review by raulsntos — APPROVED** [review]

**Pull request merged**
